src/AlarmClock.h

```diff
@@ -0,0 +1,140 @@
+/* 
+ * File:   AlarmClock.h
+ * Author: Craig Cogdill
+ * Created: October 15, 2015 10:45am
+ */
+
+#pragma once
+#include <chrono>
+#include <thread>
+#include <atomic>
+#include <future>
+#include "StopWatch.h"
+
+
+template<typename Duration> class AlarmClock {
+public:
+   typedef std::chrono::microseconds microseconds;
+   typedef std::chrono::milliseconds milliseconds;
+   typedef std::chrono::seconds seconds;
+   
+   AlarmClock(unsigned int sleepDuration) : mExpired(false),
+      kSleepTime(sleepDuration),
+      kSleepTimeMs(ConvertToMilliseconds(Duration(kSleepTime))),
+      kSleepTimeUs(ConvertToMicroseconds(Duration(kSleepTime))),
+      mKeepRunning(new bool(true)),
+      mExited(std::async(std::launch::async,
+                         &AlarmClock::AlarmClockThread,
+                         this,
+                         mKeepRunning)) {
+   }
+   
+   virtual ~AlarmClock() {
+      mKeepRunning = false;
```

Comment:
Can ```mKeepRunning``` be removed?

Instead just set ```mExpired.store(true)``` if the destructor is triggered. There is no one to read the value anyways at this point se we only need one member variable to decide the logic. 

Having atomic, as it is now, also removes any potential issues with race.